### ISSUE TITLE
Add tox EQ and improve the settings of the codec.

### ISSUE THREAD

**Issue by torvn77** [qTox/qTox#2077]
After half an hour of chatting with friends via headphones I first got sick ears,
and by the end of the hour, sick and head.
The reason is the high frequency distortions and harmonics.
Please add 
1)the EQ to filter out the muck and adjust the sound settings under the current situation.

2)higher quality codec, even though he will generate a little more traffic.

**Comment by agilob** [comment]
this is toxcore issue. please refer to https://github.com/irungentoo/toxcore and close this issue.

**Comment by torvn77** [comment]
Placed.
https://github.com/irungentoo/toxcore/issues/1404
But how close ?

**Comment by cebe** [comment]
>  But how close ?

there is a button below the comment box.

**Comment by torvn77** [comment]
Now they tell me that I did right that opened the issue here.
I will not until close, decide what will work.

I explained that the issue requires working with different parts,
EQ had here in qtox, codec in toxcore. 
This IMHO may issue remains as is to be clear what it is.
(For me as a user this one problem)

**Comment by zetok** [comment]
Eh.

@torvn77 thing is, the problem you've described is rather vague, and what you're asking for may, or may not be needed to solve problems you're experiencing.

Depending on the root problem, the issue may be with either toxcore or qTox.

> The reason is the high frequency distortions and harmonics.

That's not exactly the problem - I've have had crystal clear audio conversations. Though, in my case good microphones and net connections were used..

Problem here is that you, or the one with whom you've been talking with have some issues with network - and that would be a core issue.

What may be less likely is that things in qTox changed since I've last used it for audio conversations, which would result in audio quality being bad.

Additionally, there's a transition to new API going on - qTox still has to support new av API - without it qTox won't be able to get any fixes regarding audio quality…

**Comment by torvn77** [comment]
I realized that the improvement of the codec should ask in toxcore,
But even if the codec is good, we still would like to be able to use the EQ.
By codec there are plenty of opportunities to put pressure on my ears, since the internal mechanical noise of the laptop, thermal noise of the microphone, if you have to make it a great gain, ending with a high voice or a noise environment.
All this may cause a desire to tweak the sound with an equalizer.
With all sorts of filters on the sender's side, the sound is better not to use,
as other people difficult to understand what and how to configure.

**Comment by agilob** [comment]
qtox has a built-in audio filtering system. If enabling this setting doesn't help, then it's issue for toxcore.

![qtox](https://cloud.githubusercontent.com/assets/2544251/9293343/d419996a-441f-11e5-957f-a9d610f451b6.png)

**Comment by torvn77** [comment]
Option "Filter Audio" improves the sound, then the spoils.
I prefer to keep it disabled.

**Comment by sudden6** [comment]
I use qTox for calls and never had problems with the sound quality if a good microphone is used.
Therefore in my opinion it is out of the scope of a tox client to provide an equalizer and an external program like the pulseaudio equalizer plugin should be used.
Or if you use windows try this: http://sourceforge.net/projects/equalizerapo/

**Comment by torvn77** [comment]
The proposal to build the pulseaudio equalizer has two disadvantages:
1)Required qualification is not at all users, including me.
2)Such an equalizer will affect all the sounds from the computer.
When what are the best settings for keeping the conversation will be much different about the settings for the hearing of music.
To adjust the EQ from one setting to another, whenever the call is IMHO not be convenient.

About the microphone:
The I have a good microphone and my interlocutors to the quality coming from my sound claims do not show.
Yes and IMHO interlocutors microphones is not too bad.
Another thing is the quality of the microphone inputs on sound cards and motherboards.
Well, as it turns out, I send them a new computer?

**Comment by linux-modder** [comment]
@torvn77, 
1)  the  default  install of  pulseaudio in MOST if  not all OSes should allow enough  functionality to  change at that level.
2)  You  can  select  in playback / capture to  set separate  levels and configurations  per  'stream'  if this is not  enough consider  pulseEQ & pulse Volume meter(playback / Capture) (available in most  every  OS  i've encountered)

**Comment by agilob** [comment]
@antis81 can be closed or re-tested?

**Comment by antis81** [comment]
@agilob Please leave the issue open as at least one core part is still valid.

Tox uses Opus codec for transportation (which is totally fine). Unfortunately Tox doesn't make use of the codec's features (like e.g. echo compensation, equalizer, etc.). I'll definitely have a look into Tox core to get some information of what really goes on here. AFAIK the recorded data is in best case compressed, but otherwise not altered. Adjustment/Extension of the Tox audio transportation layer should work fine.

@torvn77 

> After half an hour of chatting with friends via headphones I first got sick ears,
> and by the end of the hour, sick and head.

Yeah, I also noticed this - especially after implementing the gain control, which amplifies the signal equally for all frequencies within range from -30~30dB. For some reason, lower freq's are almost cut out, while higher freq's are being recorded just fine. You can try this for yourself by speaking with higher/lower voice (alter the voice pitch).

However gain/volume control has now improved, making qTox audio available to the real-world. :smile:

**Comment by sudden6** [comment]
A setting for the codec bitrate has been implemented, a full EQ was determined to be out of scope.
